app/controllers/friendship_acknowledgements_controller.rb

```diff
@@ -1,26 +1,37 @@
 class FriendshipAcknowledgementsController < ApplicationController
+  before_action :authenticate_user!
```

Comment:
If you need this in most controllers you can push it up to `ApplicationController`, remove it from the child ones that declare it, and `skip_before_action` on the (supposedly) few that need it not.